docs/guides/advanced-routing.md

```diff
@@ -0,0 +1,332 @@
+    If you want both conditions to be required, you can define headers and queryParams in the same match object.
+
+[roles-and-personas]: https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1
+
+### Send traffic to coffee
+
+Using the external IP address and port for NGINX Kubernetes Gateway, we can send traffic to our coffee
+applications.
+
+> Note: If you have a DNS record allocated for `cafe.example.com`, you can send the request directly to that
+> hostname, without needing to resolve.
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+```
+
+This request should receive a response from the server `coffee-v1`.
+
+If we want our request to be routed to `coffee-v2`, then we need to meet the defined conditions. We can include
+a header:
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee -H "version:v2"
+```
+
+or include a query parameter:
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee?TEST=v2
+```
+
+Either request should result in a response from the server `coffee-v2`.
+
+## Tea applications
+
+Let's deploy a different set of applications now called `tea` and `tea-post`. These applications will
+have their own set of rules, but will still attach to the same Gateway listener as the `coffee` apps.
+
+### Deploy the tea applications
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-kubernetes-gateway/main/examples/advanced-routing/tea.yaml
+```
+
+### Deploy the HTTPRoute for the tea Services
+
+We are reusing the previous Gateway for these applications, so all we need to create is the HTTPRoute.
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: tea
+spec:
+  parentRefs:
+  - name: cafe
+  hostnames:
+  - cafe.example.com
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /tea
+      method: POST
+    backendRefs:
+    - name: tea-post-svc
+      port: 80
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /tea
+      method: GET
+    backendRefs:
+    - name: tea-svc
+      port: 80
+EOF
+```
+
+The properties of this HTTPRoute include:
+
+- The same Gateway is referenced as before.
+- The same hostname is used as with the `coffee` apps.
+- The first rule defines that a POST request to the `/tea` path is routed to the `tea-post` Service.
+- The second rule defines that a GET request to the `/tea` path is routed to the `tea` Service.
+
+### Send traffic to tea
+
+Using the external IP address and port for NGINX Kubernetes Gateway, we can send traffic to our tea
+applications.
+
+> Note: If you have a DNS record allocated for `cafe.example.com`, you can send the request directly to that
+> hostname, without needing to resolve.
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea
```

Comment:
in the other guide, we show the output from the backend. I wonder if shall be consistent.